src/Microsoft.Diagnostics.Monitoring/Configuration/AggregateSourceConfiguration.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Diagnostics.NETCore.Client;
+
+namespace Microsoft.Diagnostics.Monitoring.Configuration
+{
+    public sealed class AggregateSourceConfiguration : MonitoringSourceConfiguration
+    {
+        private IList<MonitoringSourceConfiguration> _configurations = new List<MonitoringSourceConfiguration>();
+
+        public void AddConfiguration(MonitoringSourceConfiguration configuration)
+        {
+            _configurations.Add(configuration);
+        }
+
+        public override IList<EventPipeProvider> GetProviders()
+        {
+            return _configurations.SelectMany(c => c.GetProviders()).ToList();
```

Comment:
This is a good first pass, but we may actually need to do some kind of deduplication here at some point. For example, what if two configurations provide the same provider, but at different keywords or levels? We should probably merge those.

Reply:
I've added a consideration for the future. As it is right now, there are no duplicate providers for the current combinations.